deploy/addons/gpu/amd-gpu-device-plugin.yaml.tmpl

```diff
@@ -0,0 +1,59 @@
+# Copyright 2024 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: amd-gpu-device-plugin
+  namespace: kube-system
+  labels:
+    k8s-app: amd-gpu-device-plugin
+    kubernetes.io/minikube-addons: amd-gpu-device-plugin
+    addonmanager.kubernetes.io/mode: Reconcile
+spec:
+  selector:
+    matchLabels:
+      k8s-app: amd-gpu-device-plugin
+  template:
+    metadata:
+      labels:
+        k8s-app: amd-gpu-device-plugin
+    spec:
+      nodeSelector:
+        kubernetes.io/arch: amd64
```

Comment:
Is this only supported on amd64?

Reply:
Yes, the amd gpu compatibility is only with amd64 from what I can tell.
https://rocm.docs.amd.com/projects/radeon/en/latest/docs/compatibility/native_linux/native_linux_compatibility.html